src/Text/Pretty/Simple.hs

```diff
@@ -669,3 +680,44 @@ pStringOpt outputOptions = renderLazy . layoutString outputOptions
 --
 -- If you don't want non-printable characters to be escaped, take a look at
 -- 'outputOptionsStringStyle' and 'StringOutputStyle'.
+
+
+-- | Straightforward modification of 'Data.Text.Prettyprint.Doc.Render.Terminal'
+-- , in order to make it actually lazy
+renderLazy :: SimpleDocStream AnsiStyle -> TL.Text
```

Comment:
I'd ideally like to upstream this to `prettyprinter`, rather than keep it here...

Reply:
See quchen/prettyprinter#176